rc/base/ctags.kak

```diff
@@ -6,44 +6,53 @@
 declare-option -docstring "list of paths to tag files to parse when looking up a symbol" \
     str-list ctagsfiles 'tags'
 
-define-command -params ..1 \
-    -shell-script-candidates %{
-        realpath() { ( path=$(readlink "$1"); cd "$(dirname "$1")"; printf "%s/%s\n" "$(pwd -P)" "$(basename "$1")" ) }
-        eval "set -- $kak_opt_ctagsfiles"
-        for candidate in "$@"; do
-            [ -f "$candidate" ] && realpath "$candidate"
-        done | awk '!x[$0]++' | # remove duplicates
-        while read -r tags; do
-            namecache="${tags%/*}/.kak.${tags##*/}.namecache"
-            if [ -z "$(find "$namecache" -prune -newer "$tags")" ]; then
-                cut -f 1 "$tags" | grep -v '^!' | uniq > "$namecache"
+define-command -docstring "ctags-search [<symbol>]: jump to a symbol's definition
+If no symbol is passed then the current selection is used as symbol name" \
+-shell-script-candidates %{
+    realpath() { ( path=$(readlink "$1"); cd "$(dirname "$1")"; printf "%s/%s\n" "$(pwd -P)" "$(basename "$1")" ) }
+    eval "set -- $kak_opt_ctagsfiles"
+    for candidate in "$@"; do
+        [ -f "$candidate" ] && realpath "$candidate"
+    done | awk '!x[$0]++' | # remove duplicates
+    while read -r tags; do
+        namecache="${tags%/*}/.kak.${tags##*/}.namecache"
+        if [ -z "$(find "$namecache" -prune -newer "$tags")" ]; then
+            cut -f 1 "$tags" | grep -v '^!' | uniq > "$namecache"
+        fi
+        cat "$namecache"
+    done
+} ctags-search -params ..1 %{ evaluate-commands %sh{
+    realpath() { ( path=$(readlink "$1"); cd "$(dirname "$1")"; printf "%s/%s\n" "$(pwd -P)" "$(basename "$1")" ) }
+    export tagname=${1:-${kak_selection}}
+    menu="${TMPDIR:-/tmp}/ctags-menu"
+    open='{'; close='}'
+    eval "set -- $kak_opt_ctagsfiles"
+    for candidate in "$@"; do
+        [ -f "$candidate" ] && realpath "$candidate"
+    done | awk '!x[$0]++' | # remove duplicates
+    while read -r tags; do
+        tagroot=$(realpath "${tags%/*}")
+        readtags -t "$tags" "$tagname" |
+        while read tag; do
+            name=$(printf "%s\n" "$tag" | cut -f 2 | sed "s:':'':g")
+            menuinfo=$(printf "%s\n" "$tag" | sed "s:.*/\^\(\s\+\)\?::;s:\(\\\$\)\?/$::;s:':'':g;s:$open:\\\\$open:g")
+            keys=$(printf "%s\n" "$tag" | sed "s:.*/\^::;s:\(\\\$\)\?/$::;s:':'''''''''''''''':g;s:<:<lt>:g;s:\\t:<c-v><c-i>:g")
+            file=$(printf "%s\n" "$name" | sed "s:'':'''''''''''''''':g")
+            tagroot=$(printf "%s\n" "$tagroot" | sed "s:':'''''''''''''''':g")
+            select=$(printf "%s\n" "$tagname" | sed "s:':'''''''''''''''':g;s:<:<lt>:g;s:\\t:<c-v><c-i>:g")
+            command="evaluate-commands '' try '''' edit ''''''''$tagroot/$file''''''''; execute-keys ''''''''/\Q$keys<ret>vcs\Q$select<ret>'''''''' '''' catch '''' echo -markup ''''''''{Error}unable to find tag'''''''' '''' ''"
```

Comment:
I didn't knew that this is possible when I've did this. In my opinion it will be essentially the same thing, but which will need more escaping in the end. This already works, and I've tested it with big enough search patterns.